src/lib.rs

```diff
@@ -28,25 +27,57 @@ pub mod storage;
 mod tracing;
 pub mod transport;
 
+use config::{BlockStorage, Config};
 use libp2p::{
     identity::{ed25519::Keypair as Ed25519Keypair, Keypair},
     PeerId,
 };
+use orbit::ProviderUtils;
 use relay::RelayNode;
 use routes::{delegate, invoke, open_host_key, relay_addr, util_routes::*};
 use std::{collections::HashMap, sync::RwLock};
+use storage::{
+    either::Either,
+    file_system::{FileSystemConfig, FileSystemStore},
+    s3::{S3BlockConfig, S3BlockStore},
+};
+
+pub type Block = OBlock<DefaultParams>;
+pub type BlockStores = Either<S3BlockStore, FileSystemStore>;
```

Comment:
I don't understand how this is better than an enum with all the variants of storage options

Reply:
Also, why use generics for `ImmutableStore` when in the end (I assume) it's always the `Either` enum?

Reply:
it is yeah usually gonna be that enum, I figured it would make it easier to implement alternatives if we/someone else wanted to, and easier to change the generic in `Orbit` if someone wants to deploy or embed it in a different context. e.g. if someone wanted to run a node in wasm for the browser using browser storage they could implement `HypotheticalBrowserSessionStore` and just change the generic for that environment to that and avoid potential complications of building for wasm with dependancies like s3 (although I couldn't concretely name any such complications). in short: probably overzealous abstraction